Source/Core/Core/IOS/FS/HostBackend/FS.cpp

```diff
@@ -552,16 +552,20 @@ ResultCode HostFileSystem::Rename(Uid uid, Gid gid, const std::string& old_path,
     }
   }
 
-  // Finally, remove the child from the old parent and move it to the new parent.
   FstEntry* new_entry = GetFstEntryForPath(new_path);
+  new_entry->name = split_new_path.file_name;
+
+  // Finally, remove the child from the old parent and move it to the new parent.
   const auto it = std::find_if(old_parent->children.begin(), old_parent->children.end(),
                                GetNamePredicate(split_old_path.file_name));
   if (it != old_parent->children.end())
   {
-    *new_entry = *it;
+    new_entry->data = it->data;
```

Comment:
Maybe it doesn't matter but this is different behavior than before because all of the `FstEntry` isn't copied.  It'd be nice to keep the copy constructor behavior.  What was the  ut failure?

Reply:
[``data``, ``children``, and ``name`` (which we don't care about because it's the old name) are all the members of this struct](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/IOS/FS/HostBackend/FS.h#L62).

Any test that attempted using Rename would crash when running it within the steamrt. I posted a valgrind output in a comment above.